src/components/Calculator.js

```diff
@@ -3,45 +3,63 @@
 /* eslint-disable react/prefer-stateless-function */
 import React from 'react';
 import './Calculator.css';
+import calculate from '../logic/calculate';
 
 class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      total: 0,
+      next: 0,
+    };
+
+    this.clickHandler = this.clickHandler.bind(this);
+  }
+
+  clickHandler(e) {
+    this.setState((previousState) => calculate(previousState, e.target.innerText));
   }
 
   render() {
+    let val = 0;
+    const { total, next } = this.state;
+    if (next) {
+      val = next;
+    } else if (total) {
+      val = total;
+    }
+
     return (
       <div className="calc-content">
-        <div className="calc-input">0</div>
+        <div className="calc-input">{val}</div>
         <div className="calc-row">
-          <button className="calc-button" type="button">AC</button>
-          <button className="calc-button" type="button">+/-</button>
-          <button className="calc-button" type="button">%</button>
-          <button className="calc-operator" type="button">÷</button>
+          <button className="calc-button" type="button" onClick={this.clickHandler}>AC</button>
+          <button className="calc-button" type="button" onClick={this.clickHandler}>+/-</button>
+          <button className="calc-button" type="button" onClick={this.clickHandler}>%</button>
+          <button className="calc-operator" type="button" onClick={this.clickHandler}>÷</button>
         </div>
         <div className="calc-row">
-          <button className="calc-button" type="button">7</button>
-          <button className="calc-button" type="button">8</button>
-          <button className="calc-button" type="button">9</button>
-          <button className="calc-operator" type="button">X</button>
+          <button className="calc-button" type="button" onClick={this.clickHandler}>7</button>
+          <button className="calc-button" type="button" onClick={this.clickHandler}>8</button>
+          <button className="calc-button" type="button" onClick={this.clickHandler}>9</button>
+          <button className="calc-operator" type="button" onClick={this.clickHandler}>X</button>
```

Comment:
- [ ] When I try to perform a calculation involving **`X`** (like 6 X 9), I get an error message: **Error: Unknown operation 'X'**. This is because you are using a capital letter X, whereas, the `operate.js` file recognizes small letter 'x'. Kindly change the 'X' here to 'x'

Reply:
Thank you.